From 507e6d7ba5796c276b4eec1ebd9369af5f612b7a Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Wed, 15 Apr 2015 21:51:45 +0300
Subject: [PATCH 080/800] drm: omapdrm: Don't get/put dispc in
 omap_crtc_flush()

commit 0193f0c74445ab0da0b8870c18ea5ef071fa1ec1 upstream

The omap_crtc_flush() function is always called with a reference to the
dispc held. Remove unnecessary calls to dispc_runtime_get() and
dispc_runtime_put().

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Signed-off-by: Tomi Valkeinen <tomi.valkeinen@ti.com>
---
 drivers/gpu/drm/omapdrm/omap_crtc.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/omapdrm/omap_crtc.c b/drivers/gpu/drm/omapdrm/omap_crtc.c
index 93c9431..656fa3a 100644
--- a/drivers/gpu/drm/omapdrm/omap_crtc.c
+++ b/drivers/gpu/drm/omapdrm/omap_crtc.c
@@ -368,8 +368,6 @@ static int omap_crtc_flush(struct drm_crtc *crtc)
 	WARN_ON(!drm_modeset_is_locked(&crtc->mutex));
 	WARN_ON(omap_crtc->vblank_irq.registered);
 
-	dispc_runtime_get();
-
 	if (dispc_mgr_is_enabled(omap_crtc->channel)) {
 		dispc_mgr_go(omap_crtc->channel);
 		omap_irq_register(crtc->dev, &omap_crtc->vblank_irq);
@@ -379,8 +377,6 @@ static int omap_crtc_flush(struct drm_crtc *crtc)
 		reinit_completion(&omap_crtc->completion);
 	}
 
-	dispc_runtime_put();
-
 	return 0;
 }
 
-- 
1.7.5.4

